src/app/store/ui/ui.slice.ts

```diff
@@ -0,0 +1,55 @@
+import { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { createSelector, createSlice } from '@reduxjs/toolkit';
+
+import { RootState } from '@app/store';
+
+export const uiSlice = createSlice({
+  name: 'ui',
+  initialState: { broadcasting: false, loading: false, submitted: false },
+  reducers: {
+    setBroadcasting: (state, action) => {
+      state.broadcasting = action.payload;
+    },
+    setLoading: (state, action) => {
+      state.loading = action.payload;
+    },
+    setSubmitted: (state, action) => {
+      state.submitted = action.payload;
+    },
+  },
+});
+
+const selectUiSlice = (state: RootState) => state.ui;
+const uiActions = uiSlice.actions;
+
+export function useUiState() {
```

Comment:
Not a big fan of this slice seeing it. These are states very specific to sending a tx. I'd suggest either renaming to represent the state `transaction-request.slice.ts`, or remove entirely if not a huge additional effort, instead hosting this state in a context around the form.